File: python/token.py
from type import Type
class Token(object):
    def __init__(self, token_type, token_value):
        super(Token, self).__init__()
        # 表驱动法 if
        d = {
            # ':': Type.colon,
            # ',': Type.comma,
            # '{': Type.braceLeft,
            # '}': Type.braceRight,
            '[': Type.bracketLeft,
            ']': Type.bracketRight,
        }
        if token_type == Type.auto:
            self.type = d[token_value]
        else:
            self.type = token_type

        self.value = token_value
    def __repr__(self):
        return '({})'.format(self.value)

File: javascript/Token.js
const {log, } = require('./utils')
const Type = require('./Type.js')
// log('type', Type)
class Token{
    constructor(type, value) {
        // 单字符符号
        const d = {
            // ':': Type.colon,
            // ',': Type.comma,
            // '{': Type.braceLeft,
            // '}': Type.braceRight,
            '[': Type.bracketLeft,
            ']': Type.bracketRight
        }
        // log('type', Type)
        if(type === Type.auto) {
            // 处理单字符符号
            this.type = d[value]
        } else {
            this.type = type
        }
        this.value = value
    }
    log() {
        log(`({${this.value}})`)
    }
}

module.exports = Token